scipy/ndimage/tests/test_filters.py

```diff
@@ -86,6 +86,33 @@ def test_valid_origins():
         assert_raises(ValueError, filter, data, 3, origin=2)
 
 
+def test_multiple_modes():
+    # Test that the filters with multiple mode cababilities for different
+    # dimensions give the same result as applying a single mode.
+    arr = np.zeros((100, 100), float)
+    arr[50, 50] = 1
+    arr[50, 0] = 1
+    arr[0, 0] = 1
+
+    mode1 = 'reflect'
+    mode2 = ['reflect', 'reflect']
```

Comment:
Yeah, the problem is that this kind of behaviour is mainly relevant to large arrays, and I did not really want to include any auxiliary arrays. I will try to think of a small 3x3 array that would show differences and that I could use for the tests.